pandas/core/arrays/datetimes.py

```diff
@@ -527,13 +596,9 @@ def astype(self, dtype, copy=True):
     # ----------------------------------------------------------------
     # ExtensionArray Interface
 
-    @property
-    def _ndarray_values(self):
-        return self._data
-
     @Appender(dtl.DatetimeLikeArrayMixin._validate_fill_value.__doc__)
     def _validate_fill_value(self, fill_value):
-        if isna(fill_value):
+        if isna(fill_value) or fill_value == iNaT:
```

Comment:
looks like this is hit when we change DatetimeBlock._holder to DatetimeArray; on principal we might want to avoid allowing iNaT here to avoid allowing that int64 value through

Reply:
Do you mean `DatetimeTZBlock`? We aren't (yet) changing DatetimeBlock._holder. That will continue to be an ndarray for now.

Reply:
Possibly.  IIRC DatetimeLikeBlockMixin defines _holder to DatetimeArray, but maybe that gets overridden in the `__mro__`.

Reply:
Hmm on master it's DatetimeIndex & TimedeltaIndex... I'll need to do some investigation.

Reply:
I think this comes down to `DatetimeLikeBlockMixin.fill_value` not matching the expected semantics for ExtensionDtype.na_value. The docs for ExtensionDtype.na_value state


```
    # na_value is the default NA value to use for this type. This is used in
    # e.g. ExtensionArray.take. This should be the user-facing "boxed" version
    # of the NA value, not the physical NA value for storage.
 ```

According to that definition, this should be `NaT`, not `iNaT`. I'm not sure what changing this would break though. OK with doing this as a followup.

Reply:
Fixed by #24605.